src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -44,16 +44,10 @@ internal class RazorSemanticTokensInfoService(
 
     private RazorSemanticTokensLegend? _razorSemanticTokensLegend;
 
-    public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, VSInternalClientCapabilities clientCapabilities)
+    [MemberNotNull(nameof(_razorSemanticTokensLegend))]
+    public void SetTokensLegend(RazorSemanticTokensLegend legend)
     {
-        _razorSemanticTokensLegend = new RazorSemanticTokensLegend(clientCapabilities);
-
-        serverCapabilities.SemanticTokensOptions = new SemanticTokensOptions
-        {
-            Full = false,
-            Legend = _razorSemanticTokensLegend.Legend,
-            Range = true,
-        };
+        _razorSemanticTokensLegend = legend;
```

Comment:
I'm not sure if this is the right approach, or if `GetSemanticTokensAsync` should take a `RazorSemanticTokensLegend`